daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionVersion.scala

```diff
@@ -46,6 +48,13 @@ object TransactionVersions
           .exists { case nf: Node.NodeFetch[_] => nf.actingParties.nonEmpty; case _ => false })
         minFetchActors
       else minVersion,
+      if (a.nodes.values
```

Comment:
I think this is also wrong -- I think the correct thing to do _always_ fill in the choice return value, and then decide whether to include it here based on the timeline.  @S11001001 ?

Reply:
Sure, makes sense too.

This means you can just drop the new argument, since version choice isn't a function of whether `exerciseResult.isDefined` in the scheme @bitonic's suggested.